perf(parser): inline escaped keyword check in advance()

Boshen opened the pull request:
## Summary

Manually inline the escaped keyword check directly into `advance()` instead of calling a separate `test_escaped_keyword()` function. This eliminates function call overhead on every token advance.

overlookmotel commented:
Nice!

<img width="802" height="438" alt="bench" src="https://github.com/user-attachments/assets/70ca0204-8aad-4f5c-ad17-c1ddd36b9b6c" />

Boshen committed:
perf(parser): inline escaped keyword check in advance() (#14408)

## Summary

Manually inline the escaped keyword check directly into `advance()` instead of calling a separate `test_escaped_keyword()` function. This eliminates function call overhead on every token advance.
